File: 1547-destination-city/destination-city.py
class Solution(object):
    def destCity(self, paths):
        """
        :type paths: List[List[str]]
        :rtype: str
        """
        outgoing_cities = set()
        destination = ""
        for path in paths:
            outgoing_cities.add(path[0])

        for path in paths:
            if path[1] not in outgoing_cities:
                destination = path[1]
                break
        
        return destination

File: 1547-destination-city/destination-city.js
/**
 * @param {string[][]} paths
 * @return {string}
 */
var destCity = function(paths) {
    let outgoing = new Set();


    for (let path of paths){
        outgoing.add(path[0]);
    }

    for (let path of paths){
        if(!outgoing.has(path[1])){
            return path[1];
        }
    }
};